codec.go

```diff
@@ -236,13 +237,21 @@ func (d Decimal) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + d.stringBigInt(true) + `"`), nil
 }
 
+// nullValue represents the JSON null value.
+var nullValue = []byte("null")
+
 // UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (d *Decimal) UnmarshalJSON(data []byte) error {
 	// Remove quotes if they exist.
 	if len(data) >= 2 && data[0] == '"' && data[len(data)-1] == '"' {
 		data = data[1 : len(data)-1]
 	}
 
+	// null value.
+	if bytes.Equal(data, nullValue) {
```

Comment:
The `bytes.Equal` function is optimized for performance and will perform much better than `bytes.Compare`.

Reply:
I'm good with this. Just add benchmark here for reference
```
goos: linux
goarch: amd64
cpu: Intel(R) Core(TM) i9-14900HX
BenchmarkByteEqual-32    	1000000000	         0.1213 ns/op	       0 B/op	       0 allocs/op
BenchmarkByteCmp-32    	247671904	         4.825 ns/op	       0 B/op	       0 allocs/op
```